oak_dice/src/evidence.rs

```diff
@@ -74,8 +83,35 @@ pub struct RootLayerEvidence {
 }
 
 impl RootLayerEvidence {
-    pub fn get_tee_platform(&self) -> Option<TeePlatform> {
-        TeePlatform::from_repr(self.tee_platform)
+    pub fn get_tee_platform(&self) -> Result<TeePlatform, String> {
+        TeePlatform::from_repr(self.tee_platform).ok_or("invalid TEE Platform value".to_string())
+    }
+
+    pub fn get_remote_attestation_report(&self) -> Result<&[u8], String> {
```

Comment:
Does this actually compile without lifetime annotations?

Reply:
surprisingly it does

Reply:
This looks like a simple lifetime elision for the compiler to figure out.

Reply:
https://doc.rust-lang.org/nomicon/lifetime-elision.html